### PR TITLE
chore: add aws actions

### PR DESCRIPTION
**Pull request by laurentaubin**
Added READMEs for all actions as well

### CODE REVIEW

File: actions/aws-cognito-info/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        shell: bash
+        run: |
+          echo "AWS Cognito Information:"
+          echo "User Pool ID: ${{ steps.cognito-info.outputs.user-pool-id }}"
+          echo "User Pool Client ID: ${{ steps.cognito-info.outputs.user-pool-client-id }}"
+```
+
+## Notes
+
+- Before using this action, ensure that you have set up the necessary AWS credentials to authenticate with the AWS ECR (e.g. by using the `configure-aws-credentials` action).
```
Comment (MPIngeno) [inline actions/aws-cognito-info/README.md:56]: What does the ECR have to do with this action? 🤔

File: actions/aws-s3-deploy-app/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        shell: bash
+        run: |
+          echo "AWS Region: ${{ inputs.aws-region }}"
+          echo "S3 Bucket: ${{ inputs.aws-s3-bucket }}"
+          echo "Destination Directory: ${{ inputs.destination }}"
+```
+
+## Notes
+
+- Before using this action, ensure that you have set up the necessary AWS credentials to authenticate with the AWS ECR (e.g. by using the `configure-aws-credentials` action).
```
Comment (MPIngeno) [inline actions/aws-s3-deploy-app/README.md:54]: Do we need to have the AWS credentials to authenticate with the ECR here (aka shared account)? I think the current account id (dev/staging/prod) is fine

File: actions/docker-deploy/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          echo "Repository: ${{ inputs.repository }}"
+          echo "Tag: ${{ inputs.tag }}"
+          echo "Dockerfile: ${{ inputs.dockerfile }}"
+```
+
+## Notes
+
+- Before using this action, ensure that you have set up the necessary credentials to authenticate with the container image registry.
+
+That's all for the "Docker Deploy" GitHub Action. If you have any further questions or need assistance, feel free to reach out!
```
Comment (MPIngeno) [inline actions/docker-deploy/README.md:60]: Forgot to remove the ChatGPT message here ;)

File: actions/docker-deploy/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        run: |
+          echo "Registry: ${{ inputs.registry }}"
+          echo "Repository: ${{ inputs.repository }}"
+          echo "Tag: ${{ inputs.tag }}"
+          echo "Dockerfile: ${{ inputs.dockerfile }}"
+```
+
+## Notes
+
+- Before using this action, ensure that you have set up the necessary credentials to authenticate with the container image registry.
```
Comment (MPIngeno) [inline actions/docker-deploy/README.md:58]: Indicating that the repository have the variable SHARED_SERVICES_ACCOUNT_ID set before using this action would be clearer.

File: actions/aws-cognito-info/README.md (hunk 1 above)
Comment (laurentaubin) [reply actions/aws-cognito-info/README.md:56]: Forgot to edit that line after copying it from another workflow, it's meant to say to set up regular aws credentials, not ECR.

I wasn't sure if that was needed though, I saw that in Greenfield we set up the aws credentials before using the cognito-info action so I assumed it was. Do you know if it's necessary or if we can use the cognito-info action without credentials ?

File: actions/docker-deploy/README.md (hunk 4 above)
Comment (laurentaubin) [reply actions/docker-deploy/README.md:58]: Do we need that for the actual action though ? From what I see in the `aws-docker-deploy` workflow, `SHARED_SERVICES_ACCOUNT_ID` is used to set the registry input, but the `docker-deploy` action can still be used if the registry is set in another way.

File: actions/aws-cognito-info/README.md (hunk 1 above)
Comment (laurentaubin) [reply actions/aws-cognito-info/README.md]: Ok yeah that makes sense. So we would still need to run `configure-aws-credentials` before `aws-cognito-info` then ?

Comment (MPIngeno) [reply actions/aws-cognito-info/README.md]: Yes

**Review by MPIngeno — APPROVED** [review]

File: actions/configure-aws-credentials/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# Configure AWS Credentials GitHub Action
+
+## Description
+
+The "Configure AWS Credentials" GitHub Action sets up AWS credentials for GitHub OIDC (OpenID Connect) to interact with AWS services using the specified AWS account and region. This action assumes the presence of a role named "AWSGitHubActionRoleForCICD," which is defined by the `@ingeno/aws-cdk-constructs` library.
```
Comment (jonathanpelletier) [inline actions/configure-aws-credentials/README.md:5]: This is now moved

```suggestion
The "Configure AWS Credentials" GitHub Action sets up AWS credentials for GitHub OIDC (OpenID Connect) to interact with AWS services using the specified AWS account and region. This action assumes the presence of a role named "AWSGitHubActionRoleForCICD," which is defined by the `@ingeno/foundation-infra-aws` library.
```

File: actions/configure-aws-credentials/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        shell: bash
+        run: |
+          echo "AWS Account ID: ${{ inputs.aws-account-id }}"
+          echo "AWS Region: ${{ inputs.aws-region }}"
+```
+
+## Notes
+
+- Before using this action, ensure that you have set up the necessary AWS credentials with sufficient permissions to assume the role "AWSGitHubActionRoleForCICD."
+- The action uses the `aws-actions/configure-aws-credentials@v2` action to set up the AWS credentials for OIDC.
```
Comment (jonathanpelletier) [inline actions/configure-aws-credentials/README.md:51]: Seems to be v3 in fact

```suggestion
- The action uses the `aws-actions/configure-aws-credentials@v3` action to set up the AWS credentials for OIDC.
```

Comment (jonathanpelletier) [reply actions/configure-aws-credentials/README.md:51]: Ah, never mind, this is the AWS' action... 🤦

File: actions/docker-deploy/action.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    description: Name of the repository
+
+  tag:
+    required: false
+    default: dev
+    description: Tag of the built image
+
+  dockerfile:
+    required: false
+    default: dockerfile.yml
```
Comment (jonathanpelletier) [inline actions/docker-deploy/action.yml:20]: I know this default value is coming from aws-workflows, but it should be `Dockerfile` to match how 99% of all dockerfiles are named on earth :/ If you have other changes to make around here, would be nice to update that 😉

**Comment by jonathanpelletier** [comment]
Pretty much all versions of ingeno/foundation-github-actions/** in the md files are tagged v3 :/ Now that we moved the action, the only working tag would be v1, but bumping versions to keep this up-to-date is going to be tedious. There might not be actions to be taken at this point, but maybe we need to think about our strategy here: used the "latest" tag, knowing it doesn't exist so that users replace this with an existing tag when needed or keep this updated as we increment the versions.

**Comment by laurentaubin** [comment]
I can put something like `action@<latest>` with a note on how to figure out what the latest version is.

**Comment by jonathanpelletier** [comment]
Sounds good to me 👍

**Comment by laurentaubin** [comment]
@jonathanpelletier https://github.com/ingeno/foundation-github-actions/pull/4

**Pull request merged**